Clean up `pg_tde_get_tde_data_dir()`

jeltz opened the pull request:
Make sure the function returns a const pointer and make the name consistent with `pg_tde_set_data_dir()`.

jeltz committed:
Clean up pg_tde_get_tde_data_dir()

Make sure the function returns a const pointer and make the name
consistent with pg_tde_set_data_dir().

AndersAstrand commented on contrib/pg_tde/src/common/pg_tde_utils.c:
```diff
@@ -106,8 +106,8 @@ pg_tde_set_data_dir(const char *dir)
 }
 
 /* returns the palloc'd string */
```
The string is not palloc'd, it's static, right?

jeltz replied:
Fixed

jeltz committed:
Remove incorrect comment about string being palloced

As far as I can tell this comment has never been true but was just
accidentally copied from some other fucntion.
